diff --git a/.gitlab-ci.yml b/.gitlab-ci.yml
new file mode 100644
index 0000000..e240cc2
--- /dev/null
+++ b/.gitlab-ci.yml
@@ -0,0 +1,17 @@
+# Use the DrupalCI templates.
+include:
+  - project: $_GITLAB_TEMPLATES_REPO
+    ref: $_GITLAB_TEMPLATES_REF
+    file:
+      - '/includes/include.drupalci.main.yml'
+      - '/includes/include.drupalci.variables.yml'
+      - '/includes/include.drupalci.workflows.yml'
+
+#
+# Start custom overrides.
+#
+variables:
+  # Broaden test coverage.
+  OPT_IN_TEST_PREVIOUS_MAJOR: 1
+  OPT_IN_TEST_NEXT_MINOR: 1
+  OPT_IN_TEST_MAX_PHP: 1
diff --git a/config_override_warn.info.yml b/config_override_warn.info.yml
index e248579..9d46cd9 100644
--- a/config_override_warn.info.yml
+++ b/config_override_warn.info.yml
@@ -1,4 +1,4 @@
 name: Config Override Warn
 description: Provide a warning message on configuration forms when any values on the form are possibly being overridden.
 type: module
-core_version_requirement: ^9 || ^10
+core_version_requirement: ^10.1 || ^11
diff --git a/config_override_warn.module b/config_override_warn.module
index 207420e..5325c77 100644
--- a/config_override_warn.module
+++ b/config_override_warn.module
@@ -29,7 +29,7 @@ function config_override_warn_form_alter(&$form, FormStateInterface $form_state,
       '#theme' => 'config_override_warn_overrides',
       '#overrides' => $overrides,
     ];
-    $warning = \Drupal::service('renderer')->renderPlain($override_output);
+    $warning = \Drupal::service('renderer')->renderInIsolation($override_output);
     \Drupal::service('messenger')->addWarning($warning);
   }
 }
diff --git a/tests/src/Kernel/FormOverridesTest.php b/tests/src/Kernel/FormOverridesTest.php
index 1b894de..7e9616f 100644
--- a/tests/src/Kernel/FormOverridesTest.php
+++ b/tests/src/Kernel/FormOverridesTest.php
@@ -17,10 +17,13 @@ class FormOverridesTest extends KernelTestBase {
   /**
    * {@inheritdoc}
    */
-  public static $modules = [
+  protected static $modules = [
     'system',
+    'path_alias',
     'config',
     'block',
+    'block_content',
+    'user',
     'config_override_test',
     'config_override_warn',
   ];
@@ -28,10 +31,11 @@ class FormOverridesTest extends KernelTestBase {
   /**
    * {@inheritdoc}
    */
-  protected function setUp() {
+  protected function setUp(): void {
     parent::setUp();
     $this->installEntitySchema('block');
-    $this->installConfig('config_override_test');
+    $this->installEntitySchema('block_content');
+    $this->installConfig(['config_override_test']);
   }
 
   /**
@@ -69,7 +73,7 @@ class FormOverridesTest extends KernelTestBase {
    * @return array
    *   An array of test cases.
    */
-  public function providerFormOverrides() {
+  public static function providerFormOverrides() {
     return [
       // Test with show_values turned on, show that both name and slogan are
       // overridden with their original and overridden values.
@@ -88,7 +92,7 @@ class FormOverridesTest extends KernelTestBase {
           'system.site' => [
             'name' => [
               'original' => '\'Original name\'',
-              'override' => '\'ZOMG overridden site name\'',
+              'override' => '\'Wow overridden site name\'',
             ],
             'slogan' => [
               'original' => '\'Original slogan\'',
@@ -114,7 +118,7 @@ class FormOverridesTest extends KernelTestBase {
           'system.site' => [
             'name' => [
               'original' => '\'Original name\'',
-              'override' => '\'ZOMG overridden site name\'',
+              'override' => '\'Wow overridden site name\'',
             ],
           ],
         ],
@@ -137,7 +141,7 @@ class FormOverridesTest extends KernelTestBase {
       [
         [
           'system.site' => [
-            'name' => 'ZOMG overridden site name',
+            'name' => 'Wow overridden site name',
           ],
         ],
         SiteInformationForm::class,
@@ -196,7 +200,7 @@ class FormOverridesTest extends KernelTestBase {
    * @return array
    *   An array of test cases.
    */
-  public function providerConfigEntityOverrides() {
+  public static function providerConfigEntityOverrides() {
     return [
       // Test with show_values turned on.
       [
